It worked to change the Drupal password but failed to update the live secrets file:
```
~/isle-dc$ make set_admin_password
Password:

Setting admin password now
 [success] Changed password for admin.
/bin/sh: 1: cannot create secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD: Permission denied
make: *** [Makefile:394: set_admin_password] Error 2
```
Are we expected to run the make command as root?

--- a/Makefile
+++ b/Makefile
@@ -385,13 +385,18 @@ help:
 	} \
 	{lastLine = $$0}' $(MAKEFILE_LIST)
 
+IS_DRUPAL_PSSWD_FILE_READABLE := $(shell test -r secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD -a -w secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD && echo 1 || echo 0)
+CMD := $(shell [ $(IS_DRUPAL_PSSWD_FILE_READABLE) -eq 1 ] && echo 'tee' || echo 'sudo -k tee')
+
 .PHONY: set_admin_password
 .SILENT: set_admin_password
+## Sets the admin password and accomodates for permissions restrictions to the secrets directory. Only runs sudo when needed.
 set_admin_password:
-	@$(eval PASSWORD ?= $(shell bash -c 'read -s -p "Password: " pwd; echo $$pwd'))
+	@$(eval PASSWORD ?= $(shell bash -c 'read -s -p "New Password: " pwd; echo $$pwd'))
 	@echo "\n\nSetting admin password now"
 	docker-compose exec -T drupal with-contenv bash -lc 'drush user:password admin "$(PASSWORD)"'
-	@echo "$(PASSWORD)" > secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD
+	echo "$(PASSWORD)" | $(CMD) secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD >> /dev/null
+	@echo "\ndone."
 
 LATEST_VERSION := $(shell curl -s https://api.github.com/repos/desandro/masonry/releases/latest | grep '\"tag_name\":' | sed -E 's/.*\"([^\"]+)\".*/\1/')
 

